service/resource/deployment/delete.go

```diff
@@ -84,3 +86,44 @@ func (r *Resource) newDeleteChange(ctx context.Context, obj, currentState, desir
 
 	return deploymentsToDelete, nil
 }
+
+// newDeleteChangeForUpdatePatch is used on update events to scale down
+// deployments.
+func (r *Resource) newDeleteChangeForUpdatePatch(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	currentDeployments, err := toDeployments(currentState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	desiredDeployments, err := toDeployments(desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.Log("cluster", key.ClusterID(customObject), "debug", "finding out which deployments have to be deleted")
+
+	var deploymentsToDelete []*v1beta1.Deployment
+
+	for _, currentDeployment := range currentDeployments {
+		if !containsDeployment(desiredDeployments, currentDeployment) {
+			deploymentsToDelete = append(deploymentsToDelete, currentDeployment)
+		}
+	}
+
+	r.logger.Log("cluster", key.ClusterID(customObject), "debug", fmt.Sprintf("found %d deployments that have to be deleted", len(deploymentsToDelete)))
+
+	return deploymentsToDelete, nil
+}
+
+func newDeleteOptions() *apismetav1.DeleteOptions {
+	propagation := apismetav1.DeletePropagationForeground
+
+	options := &apismetav1.DeleteOptions{
+		PropagationPolicy: &propagation,
```

Comment:
This option is very important because without it replicasets and pods would remain running when deleting a deployment. Again pointer magic because of the Kubernetes client API.